biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLQuery.java

```diff
@@ -0,0 +1,270 @@
+            for (Element hitElement : hitList) {
+                Element hitaccessionElement = XMLHelper.selectSingleElement(hitElement, "Hit_accession");
+                String hitaccession = hitaccessionElement.getTextContent();
+                Element hithspsElement = XMLHelper.selectSingleElement(hitElement, "Hit_hsps");
+                ArrayList<Element> hspList = XMLHelper.selectElements(hithspsElement, "Hsp");
+                for (Element hspElement : hspList) {
+                    Element evalueElement = XMLHelper.selectSingleElement(hspElement, "Hsp_evalue");
+                    String value = evalueElement.getTextContent();
+                    double evalue = Double.parseDouble(value);
+                    if (evalue <= maxEScore) {
+                        ArrayList<String> hits = hitsHashMap.get(querydef);
+                        if (hits == null) {
+                            hits = new ArrayList<String>();
+                            hitsHashMap.put(querydef, hits);
+                        }
+                        hits.add(hitaccession);
+                    }
+                }
+            }
+        }
+
+        return hitsHashMap;
+    }
+    
+    private void readFile(String blastFile) throws Exception{
+        log.info("Start reading " + blastFile);
+        blastDoc = XMLHelper.loadXML(blastFile);
+        log.info("Read finished");
+    }
+    
+    public ArrayList<Result> createObjects(double maxEScore) throws ParseException {
+        if (targetFile == null) throw new IllegalStateException("File to be parsed not specified.");
+        
+        try {
+            // getAbsolutePath throws SecurityException
+            readFile(targetFile.getAbsolutePath());
+        } catch (Exception e){
+            throw new ParseException(e.getMessage(),0);
+        }
+        // create mappings between sequences and blast id
+        mapIds();
+        
+        ArrayList<Result> resultsCollection;
+        ArrayList<Hit> hitsCollection;
+        ArrayList<Hsp> hspsCollection;
+        
+        try {
+            // select top level elements
+            String program = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_program").getTextContent();
+            String version = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_version").getTextContent();
+            String reference = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_reference").getTextContent();
+            String dbFile = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_db").getTextContent();
+
+            log.info("Query for hits");
+            ArrayList<Element> IterationsList = XMLHelper.selectElements(blastDoc.getDocumentElement(), "BlastOutput_iterations/Iteration[Iteration_hits]");
+            log.info(IterationsList.size() + " results");
+
+            resultsCollection = new ArrayList();

```

Comment:
It is not compulsory since I previously have declared ArrayList<Result> resultsCollection. 
Sentences like resultsCollection.add(new String()) will already fail at compilation time. I  usually prefer it, because is less verbose and in my opinion cleaner. But it should be the same. 


2015-09-22 20:57 GMT+02:00 Jose Manuel Duarte <notifications@github.com>:
> In biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLQuery.java:
> 
> > +        ArrayList<Result> resultsCollection;
> > +        ArrayList<Hit> hitsCollection;
> > +        ArrayList<Hsp> hspsCollection;
> > +        
> > +        try {
> > +            // select top level elements
> > +            String program = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_program").getTextContent();
> > +            String version = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_version").getTextContent();
> > +            String reference = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_reference").getTextContent();
> > +            String dbFile = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_db").getTextContent();
> > +
> > +            log.info("Query for hits");
> > +            ArrayList<Element> IterationsList = XMLHelper.selectElements(blastDoc.getDocumentElement(), "BlastOutput_iterations/Iteration[Iteration_hits]");
> > +            log.info(IterationsList.size() + " results");
> > +
> > +            resultsCollection = new ArrayList();
> This should be new ArrayList<Result>. In general try to parameterize classes that require them and keep the warnings to a minimum. There are a few easily avoidable warnings in this class and others (mostly related to generics parameterization).
> 
> —
> Reply to this email directly or view it on GitHub.
>